x-pack/plugins/security_solution/public/common/components/drag_and_drop/helpers.ts

```diff
@@ -197,6 +199,13 @@ export const addFieldToTimelineColumns = ({
   const fieldId = getFieldIdFromDraggable(result);
   const allColumns = getAllFieldsByName(browserFields);
   const column = allColumns[fieldId];
+  let initColumnHeader = {};
```

Comment:
nit:  since `initColumnHeader` isn't mutated after initialization, consider

```
  const initColumnHeader =
    timelineId === TimelineId.detectionsPage || timelineId === TimelineId.detectionsRulesDetailsPage
      ? alertsHeaders.find((c) => c.id === fieldId) ?? {}
      : {};
```